src/vm/method.cpp

```diff
@@ -3786,17 +3786,9 @@ void MethodDesc::SaveChunk::Append(MethodDesc * pMD)
     // Determine optional slots that are going to be saved
     if (method.m_fHasPrecode)
     {
-        method.m_fHasNativeCodeSlot = pMD->MayHaveNativeCode();
+        method.m_fHasNativeCodeSlot = TRUE;
```

Comment:
Not sure about this - discussing in parallel comments

Reply:
I removed it, I ran crossgen under a debugger and it was never hit (as we expected). I don't know what fixed the failing tests, we'll see if they continue to pass once it's back in.